homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -211,13 +211,18 @@ async def _async_update_others(self) -> None:
             log_update_error(_LOGGER, "update", self.name, "binary sensor", error)
             return
 
-        if (event_code := self.entity_description.event_code) is None:
+        if (event_codes := self.entity_description.event_codes) is None or len(
+            event_codes
+        ) == 0:
             _LOGGER.error("Binary sensor %s event code not set", self.name)
             return
 
         try:
-            self._attr_is_on = (
-                len(await self._api.async_event_channels_happened(event_code)) > 0
+            self._attr_is_on = any(
```

Comment:
I tried first doing this with a generator rather than a list comprehension, but then mypy complains with:
```
homeassistant/components/amcrest/binary_sensor.py:221: error: Argument 1 to "any" has incompatible type "AsyncGenerator[bool, None]"; expected "Iterable[object]"  [arg-type]
```
Let me try the generator instead, and silence mypy if needed...